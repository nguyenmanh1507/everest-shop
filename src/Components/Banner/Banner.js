// @flow

import React, { PureComponent } from 'react'

type Props = {
  /** Component background image */
  bgImage: string,
  /** Component title */
  title: string,
  /** Component sub title */
  subTitle: string
}

class Banner extends PureComponent<Props> {
  render() {
    const { bgImage, title, subTitle } = this.props

    return (
      <section
        className="banner"
        style={{ backgroundImage: `url(${bgImage})` }}
      >
        <div className="banner__body">
          <h3 className="banner__sub-title">{subTitle}</h3>
          <h2 className="banner__title">{title}</h2>
        </div>
      </section>
    )
  }
}

export default Banner
