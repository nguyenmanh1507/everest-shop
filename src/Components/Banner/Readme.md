Banner example:

```js
<Banner 
  title="Flowe desk"
  subTitle="Inspiration"
  bgImage="https://unsplash.it/800/450?image=230"
/>
```
