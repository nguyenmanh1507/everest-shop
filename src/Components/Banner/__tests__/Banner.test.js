// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import { render } from 'enzyme'

import Banner from '../Banner'

test('Banner render properly', () => {
  const component = renderer.create(
    <Banner
      title="Flowe desk"
      subTitle="Inspiration"
      bgImage="https://unsplash.it/800/450?image=230"
    />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
