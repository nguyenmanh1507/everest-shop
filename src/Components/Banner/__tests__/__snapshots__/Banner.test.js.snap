// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner render properly 1`] = `
<section
  className="banner"
  style={
    Object {
      "backgroundImage": "url(https://unsplash.it/800/450?image=230)",
    }
  }
>
  <div
    className="banner__body"
  >
    <h3
      className="banner__sub-title"
    >
      Inspiration
    </h3>
    <h2
      className="banner__title"
    >
      Flowe desk
    </h2>
  </div>
</section>
`;
